tests/indices/embedding/test_base.py

```diff
@@ -103,3 +108,42 @@ def test_embedding_query(
     query_str = "What is?"
     response = tree.query(query_str, mode="embedding", **query_kwargs)
     assert response == ("What is?:Hello world.")
+
+
+@patch.object(LLMChain, "predict", side_effect=mock_llmchain_predict)
+@patch.object(OpenAI, "__init__", return_value=None)
+@patch.object(
+    GPTTreeIndexEmbeddingQuery,
+    "_get_query_text_embedding_similarities",
+    side_effect=_get_node_text_embedding_similarities,
+)
+def test_query_and_count_tokens(
+    _mock_similarity: Any,
+    _mock_init: Any,
+    _mock_predict: Any,
+    struct_kwargs: Dict,
+    documents: List[Document],
+) -> None:
+    """Test query and count tokens."""
+    index_kwargs, query_kwargs = struct_kwargs
+    document_token_count = (
+        24  # mock_prompts.MOCK_SUMMARY_PROMPT_TMPL adds a "\n" to the document text
```

Comment:
ubernit: could you put comment in the line above, then can just do `document_token_count = 24`

Reply:
yup! done: https://github.com/jerryjliu/gpt_index/pull/85/commits/21d3a21748534c5b51fa216df30509843250ed2a